### PR TITLE
bug(host): Fix sync job failure blocking other set syncs

### PR DESCRIPTION
**Pull request by louisruch**
During the plugin_host_set_sync job, a single set failing to sync
can cause all following syncs to fail.

### CODE REVIEW

**Comment by louisruch** [comment]
@talanknight PTAL
This mainly adds a event.WriteError and continue instead of returning error after a failure during `ListHost` or `upsertAndCleanHosts`.

File: internal/host/plugin/job_set_sync.go [hunk 1]
```diff
@@ -574,5 +547,14 @@ func (r *SetSyncJob) upsertAndCleanHosts(
 		}
 	}
 
+	// update last sync time on the sets
+	i, err := r.writer.Exec(ctx, updateSyncDataQuery, []interface{}{setIds})
```
Comment (louisruch) [inline internal/host/plugin/job_set_sync.go:551]: @talanknight lets chat about this, I was originally going to add this to the transaction for the set, but we don't actually do a transaction if there are no changes...

**Comment by louisruch** [comment]
@talanknight @A-440Hz PTAL

**Review by talanknight — APPROVED** [review]
Looks like you cleaned up a lot of dead/duplicate code.  Thanks!  Looks great!

**Pull request merged**